app/models/extends/file.js

```diff
@@ -34,48 +34,74 @@ FileSchema.virtual('hrefs').get(function getHrefs() {
   return hasHref ? path.join(fileProvider, this.sha1) : undefined;
 });
 
-/**
- * Methods
- */
-FileSchema.methods = {
-  prepareForStorage() {
-    logger.info(`[${this.name}] preparing file for storage.`);
-
-    // Determine file mimetype from filename
-    this.mime_type = this.name ? mime.lookup(this.name) : mime.default_type;
-    logger.debug(`[${this.mime_type}] mime_type: ${this.mime_type}`);
-
-    // Handle encoding
-    logger.debug(`[${this.name}] used encoding: ${this.encoding}.`);
-    switch (this.encoding) {
-      case 'base64':
-        this.data = new Buffer(this.data, 'base64');
-        break;
-      default:
-        break;
-    }
+FileSchema.methods.prepareDataForStorage = function (i) { // eslint-disable-line
+  logger.info(`Preparing file (name: ${this.name}) for storage.`);
+
+  if (this.name) {
+    this.mime_type = mime.lookup(this.name);
+  }
+
+  if (!this.data) {
+    this.data = new Buffer('', 'utf8');
```

Comment:
probably same than `new Buffer()`

Reply:
Briefly tested with node 6.10.3
Seems to throw TypeError if Buffer constructor is called without parameters.

Reply:
ah eah its deprecated. `Buffer.alloc(0)` instead are valid..
https://nodejs.org/api/buffer.html#buffer_class_method_buffer_alloc_size_fill_encoding

Reply:
Seems to work, I'll use that one then.